homeworks/01-bubble-sort/src/App.tsx

```diff
@@ -0,0 +1,183 @@
+import { ChangeEvent, Component, MouseEvent } from "react";
+
+import Header from "./components/Header";
+import Footer from "./components/Footer";
+import Main from "./components/Main";
+
+import { Col } from "./typings/index";
+
+import generateCols from "./helpers/generate_cols";
+
+import CONFIG from "./config";
+
+import "./App.css";
+
+export enum STATUS {
+  NOT_SOLVED,
+  PAUSED,
+  SORTING,
+  SOLVED,
+}
+
+type AppProps = {};
+
+type AppState = {
+  cols: Array<Col>;
+  status: STATUS;
+  progress: {
+    i: number;
+    j: number;
+  };
+  stepInterval: number;
+  stepByStepMode: boolean;
+  stepTimeout: ReturnType<typeof setTimeout> | number;
```

Comment:
Incorrect type. What is `ReturnType<typeof setTimeout>` here? I think it should be just a `number` (or maybe 'number | null' or `number | undefined` depending on your implementation.

Try `window. setTimeout ` explicitly.